scripts/staging/knn.dml

```diff
@@ -259,17 +257,21 @@ naiveKNNsearch = function(
     Qt = t(Q);
     PQt = P %*% Qt;
     P2 = rowSums (P ^ 2);
-    D = -2 * PQt + P2;
+    Q2 = rowSums (Q ^ 2);
+    D = -2 * PQt + P2 + t(Q2);
```

Comment:
Aside from whitespace, this appears to be the only true change to the algorithm.  What is the reasoning behind the change?